test/lib/mayaUsd/fileio/testAddMayaReference.py

```diff
@@ -21,130 +21,306 @@
-from mayaUSDRegisterStrings import getMayaUsdString
+import mayaUsdAddMayaReference
 
 import unittest
 
 class AddMayaReferenceTestCase(unittest.TestCase):
     '''Test Add Maya Reference.
     '''
     pluginsLoaded = False
+    mayaSceneStr = None
+    stage = None
+    kDefaultNamespace = 'simpleSphere'
 
     @classmethod
     def setUpClass(cls):
         fixturesUtils.readOnlySetUpClass(__file__, loadPlugin=False)
         if not cls.pluginsLoaded:
             cls.pluginsLoaded = mayaUtils.isMayaUsdPluginLoaded()
 
+        # Create a pure Maya scene to reference in.
+        cls.mayaSceneStr = cls.createSimpleMayaScene()
+
     @classmethod
     def tearDownClass(cls):
         standalone.uninitialize()
 
     def setUp(self):
+        # Start each test with a new scene with empty stage.
         cmds.file(new=True, force=True)
+        import mayaUsd_createStageWithNewLayer
+        self.proxyShapePathStr = mayaUsd_createStageWithNewLayer.createStageWithNewLayer()
+        self.stage = mayaUsd.lib.GetPrim(self.proxyShapePathStr).GetStage()
 
-    def createSimpleMayaScene(self):
+    @staticmethod
+    def createSimpleMayaScene():
         import os
         import maya.cmds as cmds
         import tempfile
 
+        cmds.file(new=True, force=True)
         cmds.CreatePolygonSphere()
         tempMayaFile = os.path.join(tempfile.gettempdir(), 'simpleSphere.ma')
         cmds.file(rename=tempMayaFile)
         cmds.file(save=True, force=True, type='mayaAscii')
         return tempMayaFile
 
-    def testDefineInVariant(self):
-        '''Test the "Define in Variant" options for Add Maya Reference.'''
-
-        # First we need a pure Maya scene to reference in.
-        mayaSceneStr = self.createSimpleMayaScene()
+    def testDefault(self):
+        '''Test the default options for Add Maya Reference.
 
-        # Then create an Xform and add the maya reference.
-        cmds.file(new=True, force=True)
-        import mayaUsd_createStageWithNewLayer
-        proxyShapePathStr = mayaUsd_createStageWithNewLayer.createStageWithNewLayer()
-        stage = mayaUsd.lib.GetPrim(proxyShapePathStr).GetStage()
-        primA = stage.DefinePrim('/A', 'Xform')
-        primPathStr = proxyShapePathStr + ',/A'
+        Add a Maya Reference using the defaults (no group or variant).
+        '''
+        kDefaultPrimName = mayaUsdAddMayaReference.defaultMayaReferencePrimName()
 
         # Since this is a brand new prim, it should not have variant sets.
-        primA = mayaUsd.ufe.ufePathToPrim(primPathStr)
-        self.assertFalse(primA.HasVariantSets())
+        primTestDefault = self.stage.DefinePrim('/Test_Default', 'Xform')
+        primPathStr = self.proxyShapePathStr + ',/Test_Default'
+        self.assertFalse(primTestDefault.HasVariantSets())
+
+        mayaRefPrim = mayaUsdAddMayaReference.createMayaReferencePrim(
+            primPathStr,
+            self.mayaSceneStr,
+            self.kDefaultNamespace)
```

Comment:
Each call to createMayaReferencePrim() now captures the new return value which is the Maya Reference prim created.